### ISSUE TITLE
I got an error of confusing /MD and /MT libraries running faster_rcnn_build.m

### ISSUE THREAD

**Issue by Lee52** [ShaoqingRen/faster_rcnn#122]
I have heard about it is the problem of confusing /MD libraries and /MT libraries,
But I don't know how to solve it during the mex process,
Anyway, I am using MATLAB 2016b and use pre-compiled caffe mex file.
The error info is down below.

nms_gpu_mex.cu 
mex -outdir bin nms_gpu_mex.o -L"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v6.5\lib\x64" -lcudart
Using 'Microsoft Visual C++ 2013 Professional (C)' to compile
error using mex
LIBCMT.lib(crt0dat.obj) : error LNK2005: _amsg_exit already defined in MSVCRT.lib(MSVCR120.dll)
LIBCMT.lib(crt0dat.obj) : error LNK2005: _initterm_e already defined in MSVCRT.lib(MSVCR120.dll)
LIBCMT.lib(winapisupp.obj) : error LNK2005: __crtCaptureCurrentContext already defined in MSVCRT.lib(MSVCR120.dll)
LIBCMT.lib(winapisupp.obj) : error LNK2005: __crtCapturePreviousContext already defined in
MSVCRT.lib(MSVCR120.dll)
LIBCMT.lib(winapisupp.obj) : error LNK2005: __crtTerminateProcess already defined in MSVCRT.lib(MSVCR120.dll)
LIBCMT.lib(winapisupp.obj) : error LNK2005: __crtUnhandledException already defined in MSVCRT.lib(MSVCR120.dll)
LIBCMT.lib(hooks.obj) : error LNK2005: "void __cdecl terminate(void)" (?terminate@@YAXXZ) already defined in
MSVCRT.lib(MSVCR120.dll)
LIBCMT.lib(crt0init.obj) : error LNK2005: __xi_a already defined in MSVCRT.lib(cinitexe.obj)
LIBCMT.lib(crt0init.obj) : error LNK2005: __xi_z already defined in MSVCRT.lib(cinitexe.obj)
LIBCMT.lib(crt0init.obj) : error LNK2005: __xc_a already defined in MSVCRT.lib(cinitexe.obj)
LIBCMT.lib(crt0init.obj) : error LNK2005: __xc_z already defined in MSVCRT.lib(cinitexe.obj)
LIBCMT.lib(winxfltr.obj) : error LNK2005: __CppXcptFilter already defined in MSVCRT.lib(MSVCR120.dll)
LIBCMT.lib(mlock.obj) : error LNK2005: _lock already defined in MSVCRT.lib(MSVCR120.dll)
LIBCMT.lib(mlock.obj) : error LNK2005: _unlock already defined in MSVCRT.lib(MSVCR120.dll)
   Creating library c:\faster_rcnn-master\bin\nms_gpu_mex.lib and object c:\faster_rcnn-master\bin\nms_gpu_mex.exp
LINK : warning LNK4098: defaultlib 'MSVCRT' conflicts with use of other libs; use /NODEFAULTLIB:library
LINK : warning LNK4098: defaultlib 'LIBCMT' conflicts with use of other libs; use /NODEFAULTLIB:library
c:\faster_rcnn-master\bin\nms_gpu_mex.mexw64 : fatal error LNK1169: one or more multiply defined symbols found

error in nvmex (line 48)
eval(mexCommandLine);

error in faster_rcnn_build (line 23)
   nvmex('functions/nms/nms_gpu_mex.cu', 'bin');

**Comment by 1SoSo** [comment]
@Lee52 I am running into exactly the same issue - could you finally solve it?

**Comment by Lee52** [comment]
@1SoSo Buddy, I was going to give up and try to install py-faster-rcnn, your question suddenly innovated me and I just succeeded in compilng, and here are the solutions,
Go open the following file
C:\Users\Username\AppData\Roaming\MathWorks\MATLAB\R2016b\mex_C_win64.xml
doesn't necessarily in the exact place or with the exact name
just whichever seems to be your mex configuration for compiling 'nms_gpu_mex.cu'
Change the /MD to /MT in COMPFLAGS
And reopen MATLAB and recompile!

**Comment by Lee52** [comment]
Here are the solutions: Go change the mex setup file in the system appdata folder, change it to \MT and build again in MATLAB.

**Comment by onequery** [comment]
@Lee52 Could you explain more detail? I'm trying to find the 'mex setup file' but I can't understand what is the 'mex setup file'.

**Comment by 1SoSo** [comment]
I solved it with one of the following: 
mex -ignore LIBCMT.lib
mex f.cpp -ignore LIBCMT.lib

Unfortunately I don't remember exactly where I put this command and when. The idea behind is to ignore one of the libraries - its more of a quick fix and everything works so far. Good luck!

**Comment by onequery** [comment]
@1SoSo I can't find like that command 'mex -ignore'. But you mean that It would be OK to run the faster-rcnn even if I ignore that error, right?

**Comment by Lee52** [comment]
@assess09          Usually it is in the users/Documents and Settings(older Windows version) folder, dig deeper and find AppData/Roaming/MATLAB, and try to find the mex configure set up file and change the building setup.

**Comment by June-Jo** [comment]
@Lee52 Thanks to your solution, but I cannot find that configuration set up file in AppData\Roaming\MathWorks\MATLAB folder and I'm using window 10. Do you have any other ideas?

**Comment by onequery** [comment]
@Lee52 Thank you! I solved it! I've studied only MATLAB actually so I don't understand such a process of compilation. But fortunately I solved it and understood what you said.

Here's my solution.
There are mex configure set up file named 'mex_C_win64' and 'mex_C++_win64' at
C:/User/AppData/Roaming/Mathwork/MATLAB/R2016b
I opened them with Micro Visual Studio 2015, and fix the command from '/MD' to '/MT'

Now error messages are not appeared.
